frame/staking/src/lib.rs

```diff
@@ -1846,50 +1864,52 @@ decl_module! {
 		///    minimized (to ensure less variance)
 		///
 		/// # <weight>
-		/// E: number of edges. m: size of winner committee. n: number of nominators. d: edge degree
-		/// (16 for now) v: number of on-chain validator candidates.
-		///
-		/// NOTE: given a solution which is reduced, we can enable a new check the ensure `|E| < n +
-		/// m`. We don't do this _yet_, but our offchain worker code executes it nonetheless.
-		///
-		/// major steps (all done in `check_and_replace_solution`):
-		///
-		/// - Storage: O(1) read `ElectionStatus`.
-		/// - Storage: O(1) read `PhragmenScore`.
-		/// - Storage: O(1) read `ValidatorCount`.
-		/// - Storage: O(1) length read from `SnapshotValidators`.
-		///
-		/// - Storage: O(v) reads of `AccountId` to fetch `snapshot_validators`.
-		/// - Memory: O(m) iterations to map winner index to validator id.
-		/// - Storage: O(n) reads `AccountId` to fetch `snapshot_nominators`.
-		/// - Memory: O(n + m) reads to map index to `AccountId` for un-compact.
-		///
-		/// - Storage: O(e) accountid reads from `Nomination` to read correct nominations.
-		/// - Storage: O(e) calls into `slashable_balance_of_vote_weight` to convert ratio to staked.
-		///
-		/// - Memory: build_support_map. O(e).
-		/// - Memory: evaluate_support: O(E).
-		///
-		/// - Storage: O(e) writes to `QueuedElected`.
-		/// - Storage: O(1) write to `QueuedScore`
-		///
-		/// The weight of this call is 1/10th of the blocks total weight.
+		/// All weight notes are pertaining to the case of a better solution, in which we execute the
+		/// longest code path.
+		/// Weight: 0 + 35 * v + 25 * n
+		/// State reads:
+		/// 	- Initial checks:
+		/// 		- ElectionState, CurrentEr, QueuedScore
+		/// 		- SnapshotValidators.len()
+		/// 		- ValidatorCount
+		/// 		- SnapshotValidators
+		/// 		- SnapshotNominators
+		/// 	- Iterate over nominators:
+		/// 		- compact.len() * (Nominators(who) + SlashingSpans(who)) (self vote does not have this, yet we cannot easily know)
```

Comment:
`compact.len()` means number of nominator + number of validator or number of nominations (with validator nominating themself) ?

Reply:
`compact.len()` is basically the number of assignment, which means number of nominators who are active. 

I just added these functions here and I've added some assertions to the `sp-phragmen` crate.